### DISCUSSION TITLE
Splitting off the dictionary into a separate repo

### DISCUSSION THREAD

**Discussion by sammdot** [opensteno/plover#1472] [Ideas]
Let's face it, the dictionary is rife with inconsistencies and garbage entries. We have all the issues in #400 that still need fixing, and new ones keep popping up. Ideally, we should be able to make changes to the dictionary more frequently than it takes to get any new changes to Plover main, so I suggest moving the dictionary to a separate repository.

1. This makes the history clearer, on both Plover main and the dictionary. "When was this entry added" resolves to a specific dictionary edit, rather than being tied to a Plover version.

2. Allows for more frequent changes, so that any new changes don't necessarily have to come in a new version of Plover. Especially useful for if/when the community is planning to do a large-scale refactor of the dictionary to match the theory as described in the *Art of Chording* book better.

3. We can also move to a more human-readable/editable format that's less strict than JSON, so that we can add comments to the dictionary if needed, and convert that into JSON as part of the build process. Of course, the downside would be adding another step to the build process.

**Comment by antistic** [comment]
I like this idea, for the reasons you mentioned. Would this change include splitting off the English system (the orthography rules, the orthography word list etc.) into the other repository as well? Or strictly just the dictionary json files?

**Reply by sammdot** [reply]
Just the dictionary. It would have to be a different effort to decouple orthography from the English system, if that's something we ever decide to do.

**Comment by didoesdigital** [comment]
I'd be keen to contribute to/maintain future dictionary management.

I could also scrape some commit messages from my steno-dictionaries repo to inform future dictionary comments e.g. https://github.com/didoesdigital/steno-dictionaries/commits/master/dictionaries/top-10000-project-gutenberg-words.json
